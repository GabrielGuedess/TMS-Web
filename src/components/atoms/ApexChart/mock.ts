import { type ApexOptions } from 'apexcharts';

export const mockApexOptions: ApexOptions = {
  colors: ['#3C50E0', '#80CAEE'],
  dataLabels: {
    enabled: false,
  },
  stroke: {
    width: [2, 2],
    curve: 'straight',
  },
  legend: {
    show: false,
    position: 'top',
    horizontalAlign: 'left',
  },
  yaxis: {
    min: 0,
    max: 100,
    title: {
      style: {
        fontSize: '0px',
      },
    },
  },
  grid: {
    xaxis: {
      lines: {
        show: true,
      },
    },
    yaxis: {
      lines: {
        show: true,
      },
    },
  },
  responsive: [
    {
      breakpoint: 1024,
      options: {
        chart: {
          height: 300,
        },
      },
    },
    {
      breakpoint: 1366,
      options: {
        chart: {
          height: 350,
        },
      },
    },
  ],
  markers: {
    size: 4,
    discrete: [],
    colors: '#fff',
    strokeWidth: 3,
    fillOpacity: 1,
    strokeOpacity: 0.9,
    strokeDashArray: 0,
    strokeColors: ['#3056D3', '#80CAEE'],
    hover: {
      sizeOffset: 5,
      size: undefined,
    },
  },
  chart: {
    height: 335,
    type: 'area',
    fontFamily: 'Satoshi, sans-serif',
    toolbar: {
      show: false,
    },
    dropShadow: {
      top: 10,
      blur: 4,
      left: 0,
      opacity: 0.1,
      enabled: true,
      color: '#623CEA14',
    },
  },
  xaxis: {
    type: 'category',
    axisTicks: {
      show: false,
    },
    axisBorder: {
      show: false,
    },
    categories: [
      'Sep',
      'Oct',
      'Nov',
      'Dec',
      'Jan',
      'Feb',
      'Mar',
      'Apr',
      'May',
      'Jun',
      'Jul',
      'Aug',
    ],
  },
};
