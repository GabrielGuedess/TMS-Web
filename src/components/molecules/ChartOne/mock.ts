import { type ApexOptions } from 'apexcharts';

export const mockOptions: ApexOptions = {
  colors: ['#f1581e', '#0047ab'],
  dataLabels: {
    enabled: false,
  },
  legend: {
    show: false,
    position: 'top',
    horizontalAlign: 'left',
  },
  stroke: {
    width: [2, 2],
    curve: 'straight',
    colors: ['#f1581e', '#0047ab'],
  },
  yaxis: {
    min: 0,
    max: 100,
    title: {
      style: {
        fontSize: '0px',
      },
    },
  },
  grid: {
    borderColor: '#e5e7eb',
    xaxis: {
      lines: {
        show: true,
      },
    },
    yaxis: {
      lines: {
        show: true,
      },
    },
  },
  responsive: [
    {
      breakpoint: 1024,
      options: {
        chart: {
          height: 300,
        },
      },
    },
    {
      breakpoint: 1366,
      options: {
        chart: {
          height: 350,
        },
      },
    },
  ],
  markers: {
    size: 4,
    discrete: [],
    colors: '#fff',
    strokeWidth: 3,
    fillOpacity: 1,
    strokeOpacity: 0.9,
    strokeDashArray: 0,
    strokeColors: ['#f1581e', '#0047ab'],
    hover: {
      sizeOffset: 5,
      size: undefined,
    },
  },
  chart: {
    height: 335,
    type: 'area',
    fontFamily: 'Satoshi, sans-serif',
    toolbar: {
      show: false,
    },
    dropShadow: {
      top: 10,
      blur: 4,
      left: 0,
      opacity: 0.1,
      enabled: true,
      color: '#623CEA14',
    },
  },
  xaxis: {
    type: 'category',
    axisTicks: {
      show: false,
    },
    axisBorder: {
      show: false,
    },
    categories: [
      'Sep',
      'Oct',
      'Nov',
      'Dec',
      'Jan',
      'Feb',
      'Mar',
      'Apr',
      'May',
      'Jun',
      'Jul',
      'Aug',
    ],
  },
};
