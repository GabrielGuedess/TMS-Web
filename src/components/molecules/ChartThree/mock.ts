import { type ApexOptions } from 'apexcharts';

export const mockOptions: ApexOptions = {
  series: [76, 67, 61],
  colors: ['#1ab7ea', '#0084ff', '#39539E'],
  labels: ['Desktop', 'Mobile', 'Desconhecido'],
  chart: {
    height: 390,
    type: 'radialBar',
  },
  legend: {
    show: true,
    offsetY: 15,
    offsetX: -30,
    floating: true,
    fontSize: '16px',
    position: 'left',
    itemMargin: {
      vertical: 3,
    },
    labels: {
      useSeriesColors: true,
    },
  },
  plotOptions: {
    radialBar: {
      offsetY: 0,
      startAngle: 0,
      endAngle: 270,
      hollow: {
        margin: 5,
        size: '30%',
        image: undefined,
        background: 'transparent',
      },
      dataLabels: {
        name: {
          show: false,
        },
        value: {
          show: false,
        },
      },
    },
  },
};
