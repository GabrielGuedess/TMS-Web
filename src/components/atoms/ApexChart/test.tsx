import { render, waitFor } from '@testing-library/react';

import { ApexChart } from '.';
import { mockApexOptions } from './mock';

describe('<ApexChart />', () => {
  it('should render the apex chart correctly', async () => {
    await waitFor(() => {
      const { container } = render(
        <ApexChart
          series={[
            {
              name: 'Product One',
              data: [23, 11, 22, 27, 13, 22, 37, 21, 44, 22, 30, 45],
            },
            {
              name: 'Product Two',
              data: [30, 25, 36, 30, 45, 35, 64, 52, 59, 36, 39, 51],
            },
          ]}
          type="area"
          width={100}
          height={100}
          options={mockApexOptions}
        />,
      );

      expect(container.firstChild).toMatchSnapshot();
    });
  });
});
