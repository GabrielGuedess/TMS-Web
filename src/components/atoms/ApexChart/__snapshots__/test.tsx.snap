// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ApexChart /> should render the apex chart correctly 1`] = `null`;
